Update Security Feature guide to cover ESP-IDF latest version changes. (#71)
FreeRTOS/iot-reference-esp32c3: UseSecurityFeatures.md

3. Select `Security features`.
4. Set `Enable hardware Secure Boot in bootloader (READ DOCS FIRST)` to true.
5. Set `Sign binaries during build` to true.
6. Set `Secure boot private signing key` to the path to the RSA 3072 private
key you generated in step 1.
7. Go back to main menu.
1. Select `Bootloader config`
1. Select `Bootloader log verbosity`
1. Select `Warning` or `Error`

> 1. Go back to main menu, Save and Exit.

Reviewer: Remove this,  and add this
```
# Shrink bootloader by setting the default bootloader log level to error.
CONFIG_BOOTLOADER_LOG_LEVEL_ERROR=y
```

To the `sdkconfig.default` file please.

==============================================================================
```
2. Flash the bootloader by copying and pasting the command under "Secure boot
enabled, so bootloader not flashed automatically," (the second block of text),
replacing:
    - **(PORT)** with the serial port to which the ESP32-C3 is connected. (Do not include
the opening and closing braces around PORT in the command)
    - **(BAUD)** with 460800.

> **NOTE**: If bootloader image is too big that exceeds bootloader size limit, you can [reduce bootloader log level](https://docs.espressif.com/projects/esp-idf/en/release-v4.3/esp32/security/secure-boot-bootloader-size.html). Then, rebuild the bootloader image.

Reviewer: ```suggestion
**NOTE**: If bootloader image is too big that exceeds bootloader size limit, you can
[reduce bootloader log level](https://docs.espressif.com/projects/esp-idf/en/release-v4.3/esp32/security/secure-boot-bootloader-size.html)
to reduce the size of the image. Then, rebuild the bootloader image.
```

==============================================================================
```
2. Flash the bootloader by copying and pasting the command under "Secure boot
enabled, so bootloader not flashed automatically," (the second block of text),
replacing:
    - **(PORT)** with the serial port to which the ESP32-C3 is connected.
     (Do not includethe opening and closing braces around PORT in the command)
    - **(BAUD)** with 460800.

> **NOTE**: If bootloader image is too big that exceeds bootloader size limit,

Reviewer: I think we should just change the default bootloader verbosity to ERROR or WARN instead of doing this. I don't believe the build command will tell you if the bootloader is too big or small. Where this just leads to the following flash failing, and then needing to re-do all this.

Author: We could do that.  It is just it is not happening with c3 board.

> you can [reduce bootloader log level](https://docs.espressif.com/projects/esp-idf/en/release-v4.3/esp32/security/secure-boot-bootloader-size.html). After the configuration change,

Reviewer: ```suggestion
you can [reduce bootloader log level](https://docs.espressif.com/projects/esp-idf/en/release-v5.1/esp32/security/secure-boot-v2.html#bootloader-size). After the configuration change,
```

Author: This part has been removed.

```

> idf.py -p PORT flash monitor

Reviewer: Shouldn't this be encrypted-flash here as well?
Isn't the point of the above to use encrypted-flash for this build?

Author: no, the first one does not need the encrypt in it.
The subsequent ones need it.  It is stated in the comment.

Reviewer: Got it, thanks!

I (198603) ota_over_mqtt_demo:  Received: 160   Queued: 160   Processed: 160   Dropped: 0
```

Once all the firmware image chunks are downloaded and the signature is validated,
the device reboots with the new image, during which the Secure Boot sequence is

> executed. See the OTA section in the [Featured FreeRTOS IoT Integration page for the ESP32-C3](https://www.freertos.org/featured-freertos-iot-integration-targeting-an-espressif-esp32-c3-risc-v-mcu/)

Reviewer: ```suggestion
executed. See the OTA section in the
[Featured FreeRTOS IoT Integration page for the ESP32-C3](https://www.freertos.org/featured-freertos-iot-integration-targeting-an-espressif-esp32-c3-risc-v-mcu/)
```
